lib/telegraph/node_element.rb

```diff
@@ -0,0 +1,59 @@
+module Telegraph
+  class NodeElement
+    AVAILABLE_TAGS = %w{
+      a
+      aside
+      b
+      blockquote
+      br
+      code
+      em
+      figcaption
+      figure
+      h3
+      h4
+      hr
+      i
+      iframe
+      img
+      li
+      ol
+      p
+      pre
+      s
+      strong
+      u
+      ul
+      video
+    }
+
+    AVAILABLE_ATTRIBUTES = %w{ href src }
+
+    attr_reader :tag, :attrs, :children
+
+    def initialize(tag, attrs = {}, children = [])
```

Comment:
This way (with hash in the middle) it is not convenient to call that: 
```ruby
NodeElement.new('div', {href: 'foo'}, [child1, child2])
```
WDYT about
```ruby
NodeElement.new('div', child1, child2, href: 'foo')
```
?